pydoctor/epydoc2stan.py

```diff
@@ -282,13 +282,18 @@ def __init__(self, obj: model.Documentable):
     def set_param_types_from_annotations(
             self, annotations: Mapping[str, Optional[ast.expr]]
             ) -> None:
+        """This method MUST only be called for Function instances."""
+        assert isinstance(self.obj, model.Function)
         _linker = self.obj.docstring_linker
+
         formatted_annotations = {
             name: None if value is None
-                       else ParamType(safe_to_stan(colorize_inline_pyval(value, is_annotation=True), _linker,
-                                self.obj, fallback=colorized_pyval_fallback, section='annotation', report=False),
-                                # don't spam the log, invalid annotation are going to be reported when the signature gets colorized
-                                origin=FieldOrigin.FROM_AST)
+                       else ParamType(safe_to_stan(colorize_inline_pyval(value, 
+                                 refmap=model.type_param_refs(self.obj), 
```

Comment:
There is a major problem regarding this way of resolving the type parameters. The fact that we resolve early references to type parameters and resolve lazily other references creates situations where resolving simply can't be accurate.